### ISSUE TITLE
Missing library: libc.so.6

### ISSUE THREAD

**Issue by ghost** [ValveSoftware/steam-for-linux#3730]
Ubuntu 14.04.2 desktop 64-bit (clean fresh install)
default AMD radeon open source drivers
Steam client 1.0.0.49

Steam needs to install these additional packages: 
    libgl1-mesa-dri:i386, libgl1-mesa-glx:i386, libc6:i386

[sudo] password for *****: 
.................................................................................................
Reading package lists... Done
Building dependency tree  
Reading state information... Done
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

The following packages have unmet dependencies:
 libgl1-mesa-glx:i386 : Depends: libglapi-mesa:i386 (= 10.1.3-0ubuntu0.3)
 unity-control-center : Depends: libcheese-gtk23 (>= 3.4.0) but it is not going to be installed
                        Depends: libcheese7 (>= 3.0.1) but it is not going to be installed
E: Error, pkgProblemResolver::Resolve generated breaks, this may be caused by held packages.

Press return to continue:

You are missing the following 32-bit libraries, and Steam may not run:
libc.so.6

Note: Everything was working fine on 14.04.1

**Comment by silasthefirst** [comment]
I have encountered the same missing libc.so.6 error as of yesterday.

Everything was working perfectly until the Feb 18th update.

In my case, Steam will launch, but some games are broken. For example, I can launch Borderlands 2 fine, Kingdom Rush, however, goes to a black screen, then closes.

I have purged Steam, re-installed from steampowered and from the Debian repos, checked all the symlinks, etc. Nothing has fixed the issue for me.

I am using Debian Jessie/Sid with libc6:i386 version 2.19

The only two interesting things that come up on to console are:
Error: You are missing the following 32-bit libraries, and Steam may not run:
libc.so.6

and

.steam/root/steam.sh: line 756: .steam/root/ubuntu12_32/steam: Too many levels of symbolic links

The later comes when I close Steam.

**Comment by dsx75** [comment]
Probably duplicate of #3728

**Comment by gdrewb-valve** [comment]
Assigning separately just in case these are separate issues.

**Comment by dsx75** [comment]
Probably related bug:
https://bugs.launchpad.net/ubuntu/+source/mesa/+bug/1424263

**Comment by Fincer** [comment]
Confirming, I have this issue on Arch Linux 64-bit. Appeared after the latest Steam update.

#####################################

Running Steam on arch  64-bit
STEAM_RUNTIME has been set by the user to: /home/fincer/.local/share/Steam/ubuntu12_32/steam-runtime
Error: You are missing the following 32-bit libraries, and Steam may not run:
libc.so.6

/home/fincer/.steam/root/steam.sh: line 756: /home/fincer/.steam/root/ubuntu12_32/steam: Too many levels of symbolic links

#####################################

**Comment by ekner** [comment]
I can also confirm this, Ubuntu 15.04. It wants, exactly as dsx75 says, to install these packages: libgl1-mesa-dri:i386, libgl1-mesa-glx:i386, libc6:i386

But when I try to apt-get them, it can't even find them.

**Comment by ekner** [comment]
This command solves it for me:
```sudo apt-get install '^libc6.*'```
(from http://askubuntu.com/questions/597980/error-while-installing-steam )

But the user shouldn't have to do that, so I guess this is still a bug.

**Comment by voyager2102** [comment]
Try echo $HOME and see if your $HOME ends in a slash. That was the problem with my installation - that made the check 
if [ "$STEAMROOT" != "$STEAMROOTLINK" -a "$STEAMROOT" != "$STEAMDATALINK" ]; then
fail since /my/homedir//.steam... != /my/homedir/.steam...
After removing the / in /etc/passwd Steam found its libraries again and the looping root link fixed itself.

**Comment by mdinger** [comment]
I installed these dependencies as prerequisites to building [servo](https://github.com/servo/servo) and it seems to have fixed this:

```cmd
sudo apt-get install curl freeglut3-dev \
    libfreetype6-dev libgl1-mesa-dri libglib2.0-dev xorg-dev \
    msttcorefonts gperf g++ cmake python-virtualenv \
    libssl-dev libbz2-dev libosmesa6-dev libxmu6 libxmu-dev
```

My guess is the fix was `libgl1-mesa-dri` or `libosmesa6-dev` but I'm not really sure as I didn't track it down.

**Comment by narenlegha** [comment]
I was having this problem on _newly installed_ 64-Bit Ubuntu 14.04.2 LTS.
Steam was having problem with these 3 packages:
    `libgl1-mesa-dri:i386, libgl1-mesa-glx:i386, libc6:i386`

I tried this command:
`sudo apt-get install libc6:i386 libgl1-mesa-dri-lts-utopic:i386 libgl1-mesa-glx-lts-utopic:i386` 

This removed the problem regarding missing library `libc.so.6` on my machine.

**Comment by thankjura** [comment]
@voyager2102 thanks, this work.

**Comment by takeeast** [comment]
@voyager2102 This solution works for me! Thanks! 
###### For the people who may have problems understanding the solution: 
**1.** navigate to /etc/ and edit the text-file "passwd" as root
**2.** search for your username and you will find the path to your home directory in that line
**3.** change the path from .../path to your home directory/ to .../path to your home directory (without the "/" at the end)
**4.** restart your computer

**Comment by Tele42** [comment]
Alternatively, you could you the very old, very reliable utility ```usermod -m -d /home/username username``` which protects you, the sysadmin, from typos and other forms of chaos that could render your system inaccessible.

**Comment by pilotforhire** [comment]
I can confirm @takeeast 's claim. My home directory was set to "/home/user/" and after I had fixed that to "/home/user" Steam started to work again. Conclusion: The problem is not a bug in steam but a wrong home directory string.

**Comment by voyager2102** [comment]
@pilotforhire: Your conclusion is wrong. The bug is in steam as the field you're talking about is defined as "Path to the user's home directory." or in longer form: "The home directory field provides the name of the initial working directory." In this context it does not matter if you specify a trailing slash or not and any program parsing this has to be aware of that.

The author of the script might have e.g. used readlink -f or a similar tool to canonicalize the path before using it.

**Comment by Vikloen** [comment]
I am on ubuntu 14.04.2 64bits.

1/
Install steam.deb package from this location : https://github.com/ValveSoftware/steam-for-linux = fail ...You are missing the following 32-bit libraries, and Steam may not run: libc.so.6.
![cccccccccc](https://cloud.githubusercontent.com/assets/12892023/9003614/b0e1597c-376f-11e5-9f72-994c5d3687b9.png)

2/
Install steam via Ubuntu Software Center = fail !
(https://github.com/ValveSoftware/steam-for-linux/issues/3749#issuecomment-120926246)

3/
Actually only install steam via synaptic seem to work.

**Comment by steveklabnik** [comment]
I am getting this after an update today. I have installed all of these packages, and I have a `/lib32/libc.so.6`, so it's certainly there, I just don't know why steam can't find it :/

**Comment by steveklabnik** [comment]
Furthermore, running `ldd` on the contents of `~/.steam/bin` shows no entries for it being missing, and several where it's properly linked.

**Comment by Fincer** [comment]
Thanks voyager2102!

Simply changing

`fincer:x:1000:1000::/home/fincer/:/bin/bash`

to

`fincer:x:1000:1000::/home/fincer:/bin/bash`

in `/etc/passwd` file

and rebooting the computer fixed the issue for me. At last!

**Comment by TamaHobbit** [comment]
Same problem with a fresh install of Ubuntu 14.04.1 64-bit. $HOME does not end in / so voyager's fix won't work. Trying to install libgl1-mesa-dri gives the same error about a dependency that cannot be resolved, so narenlegha and mdinger's fix doesn't work.

My fix was:
* This tells you what to put instead of "lts-vivid" for your machine

    dpkg -l | grep xserver-xorg-core

* Run the following with your substitution for lts-vivid (two places)

    sudo apt-get install xserver-xorg-lts-vivid libgl1-mesa-glx-lts-vivid:i386

So, for example, if dpkg -l | grep xserver-xorg-core gives you:

    ii  xserver-xorg-core-lts-trusty                           2:1.17.1-0ubuntu3~trusty1                           amd64        Xorg X server - core server

you should run:

    sudo apt-get install xserver-xorg-lts-trusty libgl1-mesa-glx-lts-trusty:i386

**Comment by Nerfair** [comment]
On debian install this https://packages.debian.org/jessie/amd64/libc6-i386/download

**Comment by Senjihakku** [comment]
Just tried to install Steam using the deb package. Same problem as everyone.
I use Linux Debian Jessie 8, kernel version 3.16.0-4 (64 bits) and my desktop is KDE4.
I tried literally everything stated in this page. Nothing works. The program still reports unmet dependencies:

> Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

The following packages have unmet dependencies:
 libgl1-mesa-dri:i386 : Depends: libllvm3.5:i386 but it is not going to be installed
                        Recommends: libtxc-dxtn-s2tc0:i386 but it is not going to be installed or
                                    libtxc-dxtn0:i386
E: Unable to correct problems, you have held broken packages.
Press return to continue: 

When trying to install: libgl1-mesa-dri:i386, libgl1-mesa-glx:i386, libc6:i386

Steam will not continue without these libraries.

I tried using steamcmd.sh, it says "./steamcmd.sh: line 29: /home/senjihaku/steamcmd2/linux32/steamcmd: No such file or directory".

I tried installing the libraries manually. Same issue as above (unmet dependencies).

I tried the symbolic links. It didn't work.

My passwd user entry does not end with a slash (/). It makes no difference.

**Comment by kisak-valve** [comment]
Hello @Senjihakku, you might have more success installing steam on Debian by using their guide at https://wiki.debian.org/Steam, however, the message it is giving you may also be accurate and there a package conflict that needed to be figured out and the package manager is preventing the system from becoming more broken.

**Comment by wb7odyfred** [comment]
From starting the Steam Client, I Got error MIssing library: libc.so.6  for  http://Q4OS.org  Debian 9 Stretch Linux.
`whoami`
**steam**
`su normal_User_Name`
password: xxxxx
`whoami`
**normal_User_Name**
`sudo apt-get install libc6:i386 libgl1-mesa-dri:i386 libgl1-mesa-glx:i386`

**Comment by GZGavinZhao** [comment]
On Lenovo Chromebook Duet, with Linux(beta) enabled, arm64
When I attempt to launch Steam, it tells me that I'm missing the 32-bit library `libc.so.6`, so I can't even open Steam.
I assume it's because `libc6` simply doesn't have an armV7 library or so. Maybe the problem will be solved once Steam doesn't require the 32-bit library, or when `libc6` will have a distribution for armV7.

**Comment by kisak-valve** [comment]
Hello @GZGavinZhao, Steam is not available for ARM based systems and #4061 is a better match to the issue you've encountered.

It looks like the problem tracked here was a system config issue and the reporting user has turned into a ghost. Closing.

**Comment by Impulse706** [comment]
bro i need help. anyone out there? thanks.
